File: ports/java/src/gui/painting/BcsPen.java
package bcs.gui.painting;

import bcs.core.global.BcsCoreTypes.BcsGlobalColor;

// BcsPen defines how lines and outlines are drawn
public class BcsPen {
    public BcsGlobalColor color;
    public int width;

    public BcsPen(BcsGlobalColor color, int width) {
        this.color = color;
        this.width = width;
    }
}

// BcsBrush defines how shapes are filled
class BcsBrush {
    public BcsGlobalColor color;

    public BcsBrush(BcsGlobalColor color) {
        this.color = color;
    }
}

File: ports/csharp/src/gui/painting/BcsPen.cs
using BCS.Core.Global;

namespace BCS.Gui.Painting {
    // BcsPen defines how lines and outlines are drawn
    public class BcsPen {
        public BcsGlobalColor Color { get; set; }
        public int Width { get; set; }

        public BcsPen(BcsGlobalColor color, int width = 1) {
            Color = color;
            Width = width;
        }
    }

    // BcsBrush defines how shapes are filled
    public class BcsBrush {
        public BcsGlobalColor Color { get; set; }

        public BcsBrush(BcsGlobalColor color) {
            Color = color;
        }
    }
}
